scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1060,9 +1087,11 @@ def update_cache( self, run_state ):
                 self._pods_cache.mark_as_expired(current_time)
 
                 self._update_cluster_name( local_state.k8s )
-                # TODO-163: Don't call this every time through the loop
-                # Maybe once an hour or once a day
-                self._update_api_server_version(local_state.k8s)
+
+                # Update the api server version every hour
+                if last_version_check_time is None or current_time - last_version_check_time > 3600:
```

Comment:
We could do away with the `is None` check if `last_version_check_time` was initially set to `0` (the beginning of the epoch) because `current_time - <00:00:00 1 January 1970>` is always going to be greater than 3600.

I know we had a discussion previously about not confusing semantics, but I feel this semantic is not so mixed, and it also makes the code easier to read.

I'll leave it up to you to decide.

Reply:
Good suggestion!